src/parser/ion_1_0/text/mod.rs

```diff
@@ -0,0 +1,2614 @@
+            map(take_clob_escape, |esc| esc.into_ascii_escape()),
+        )(i)
+    })(i)
+}
+
+/// The quoting strategy is taken care of by a long quoted string helper.
+///
+/// Encoding: CLOB_LONG_TEXT
+/// fragment
+/// CLOB_LONG_TEXT
+///     : CLOB_LONG_TEXT_NO_QUOTE
+///     | '\'' CLOB_LONG_TEXT_NO_QUOTE
+///     | '\'\'' CLOB_LONG_TEXT_NO_QUOTE
+///     ;
+fn take_clob_long_text(i: &[u8]) -> IonResult<&[u8], Vec<u8>> {
+    take_delimited_input(LONG_QUOTE, |i| {
+        escaped_transform(
+            take_clob_long_text_allowed,
+            COMMON_ESCAPE,
+            map(take_clob_escape, |esc| esc.into_ascii_escape()),
+        )(i)
+    })(i)
+}
+
+/// Omitted - using a shared long quoted string parser which can be specialized for bodies.
+/// See take_long_quoted_string_text_allowed
+///
+/// fragment
+/// CLOB_LONG_TEXT_NO_QUOTE
+///     : (CLOB_ESCAPE | CLOB_LONG_TEXT_ALLOWED)
+///     ;
+
+/// Parser of CLOB_SHORT_TEXT_ALLOWED
+fn take_clob_short_text_allowed(i: &[u8]) -> IonResult<&[u8], &[u8]> {
+    take_while1(is_clob_short_text_allowed)(i)
+}
+
+/// non-control ASCII and not double quote or backslash
+/// fragment
+/// CLOB_SHORT_TEXT_ALLOWED
+///     : '\u0020'..'\u0021' // no U+0022 double quote
+///     | '\u0023'..'\u005B' // no U+005C backslash
+///     | '\u005D'..'\u007F'
+///     | WS_NOT_NL
+///     ;
+fn is_clob_short_text_allowed(b: u8) -> bool {
+    is_ws_not_nl(b as char) || (b >= 0x20 && b.as_char() != '"' && b.as_char() != '\\' && b <= 0x7f)
+}
+
+/// Parser of CLOB_LONG_TEXT_ALLOWED
+fn take_clob_long_text_allowed(i: &[u8]) -> IonResult<&[u8], &[u8]> {
+    take_long_quoted_string_text_allowed(is_clob_long_text_allowed)(i)
+}
+
+/// non-control ASCII (newlines are OK)
+/// fragment
+/// CLOB_LONG_TEXT_ALLOWED
+///     : '\u0020'..'\u0026' // no U+0027 single quote
+///     | '\u0028'..'\u005B' // no U+005C blackslash
+///     | '\u005D'..'\u007F'
+///     | WS
+///     ;
+fn is_clob_long_text_allowed(b: u8) -> bool {
+    // Note: we allow the single quote and explicitly lookahead for the triple quote while iterating.
+    is_ws(b.as_char()) || (b >= 0x20 && b.as_char() != '\\' && b.as_char() != '\'' && b <= 0x7f)
+}
+
+/// fragment
+/// CLOB_ESCAPE
+///     : COMMON_ESCAPE | HEX_ESCAPE
+///     ;
+fn take_clob_escape<Input>(i: Input) -> IonResult<Input, Escape>
+where
+    Input: Clone
+        + AsBytes
+        + InputIter
+        + InputLength
+        + InputTake
+        + Compare<&'static str>
+        + Slice<Range<usize>>
+        + Slice<RangeFrom<usize>>
+        + Slice<RangeTo<usize>>,
+    <Input as InputIter>::Item: AsChar,
+{
+    cut(alt((take_common_escape_code, take_hex_escape)))(i)
+}
+
+///
+/// Encoding Section: Ion BLOB
+///
+
+/// LOB_START * LOB_END is handled by take_lob above which can be
+/// specialized with a body parser.
+///
+/// BLOB
+///     : LOB_START (BASE_64_QUARTET | WS)* BASE_64_PAD? WS* LOB_END
+///     ;
+fn take_blob_body(i: &[u8]) -> IonResult<&[u8], ion::Blo
```

Comment:
Replace "unsafe" with a different word unless you actually mean `unsafe`.

I think I may need to circle back and take a longer look at this at some point

Reply:
It's definitely not actually unsafe. The issue is that if it is applied as a normal parser, the input slice does not advance for consumed data. This could be fixed, but it is also kind of meaningless since when functioning via map_parser it is passed only the slice that was consumed by the previous parser (so the result *should* always be an empty slice which points to the last place in the input slice). I've adjusted the comment.